src/Domain/CoordinateMaps/FocallyLiftedEndcap.hpp

```diff
@@ -0,0 +1,426 @@
+ * `deriv_lambda_tilde` takes as arguments \f$x_0^i\f$, a projection point
+ *  \f$P^i\f$, and \f$\tilde{\lambda}\f$, and
+ *  returns \f$\partial \tilde{\lambda}/\partial x^i\f$.
+ * By differentiating Eq. (11), we find
+ *
+ * \f{align}
+ * \frac{\partial\tilde{\lambda}}{\partial x^j} &=
+ * \tilde{\lambda}^2 \frac{C^j - x_0^j}{|x_0^i - P^i|^2
+ * + (x_0^i - P^i)(P_i - C_{i})}.
+ * \f}
+ *
+ * ### inv_jacobian
+ *
+ * `inv_jacobian` returns \f$\partial \bar{x}^i/\partial x_0^k\f$,
+ *  where \f$\sigma\f$ is held fixed.
+ *  The arguments to `inv_jacobian`
+ *  are \f$(\bar{x},\bar{y},\bar{z})\f$, but \f$\bar{z}\f$ is ignored.
+ *
+ * Note that \f$\bar{x}\f$ and \f$\bar{y}\f$ can be considered to
+ * depend only on \f$x_0^0\f$ and \f$x_0^1\f$ but not on \f$x_0^2\f$,
+ * because the point \f$x_0^i\f$ is constrained to lie on a sphere of
+ * radius \f$R\f$.  Note that there is an alternative way to compute
+ * Eqs. (8) and (9) using only \f$x_0^0\f$ and \f$x_0^1\f$. To do
+ * this, define
+ *
+ * \f{align}
+ * \upsilon \equiv \sin(\bar{\rho}\theta_\mathrm{max})
+ *        &= \sqrt{\frac{(x_0^0-C^0)^2+(x_0^1-C^1)^2}{R^2}}.
+ * \f}
+ *
+ * Then we can write
+ *
+ * \f{align}
+ * \frac{1}{\bar{\rho}}\sin(\bar{\rho}\theta_\mathrm{max})
+ * &= \frac{\theta_\mathrm{max}\upsilon}{\arcsin(\upsilon)},
+ * \f}
+ *
+ * so that
+ *
+ * \f{align}
+ * \bar{x} &= \frac{x_0^0-C^0}{R}\left(\frac{1}{\bar{\rho}}
+ *             \sin(\bar{\rho}\theta_\mathrm{max})\right)^{-1} \\
+ * \bar{y} &= \frac{x_0^1-C^1}{R}\left(\frac{1}{\bar{\rho}}
+ *             \sin(\bar{\rho}\theta_\mathrm{max})\right)^{-1}.
+ * \f}
+ *
+ * We will compute \f$\partial \bar{x}^i/\partial
+ * x_0^k\f$ by differentiating Eqs. (15) and (16).  Because those equations
+ * involve \f$\bar{\rho}\f$, we first establish some relations
+ * involving derivatives of \f$\bar{\rho}\f$.  For ease of notation, we define
+ *
+ * \f[
+ * q \equiv \frac{\sin(\bar{\rho}\theta_\mathrm{max})}{\bar{\rho}}.
+ * \f]
+ *
+ * First observe that
+ * \f[
+ * \frac{dq}{d\upsilon}
+ * = \frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},
+ * \f]
+ *
+ * where \f$\upsilon\f$ is the quantity defined by Eq. (13).  Therefore
+ *
+ * \f{align}
+ * \frac{\partial q}{\partial x_0^0} &=
+ * \frac{\bar{x}}{\bar{\rho}R}\frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},\\
+ * \frac{\partial q}{\partial x_0^1} &=
+ * \frac{\bar{y}}{\bar{\rho}R}\frac{dq}{d\bar{\rho}}
+ * \left(\bar{\rho} \frac{dq}{d\bar{\rho}} + q\right)^{-1},
+ * \f}
+ *
+ * where we have differentiated Eq. (13), and where we have
+ * used Eqs. (15) and (16) to eliminate \f$x_0^0\f$ and
+ * \f$x_0^1\f$ in favor of \f$\bar{x}\f$ and
+ * \f$\bar{y}\f$ in the final result.
+ *
+ * By differentiating Eqs. (15) and (16), and using Eqs. (17) and (18), we
+ * find
+ *
+ * \f{align*}
+ * \frac{\partial \bar{x}}{\partial x_0^0} &=
+ * \frac{1}{R q}
+ * - \frac{\bar{x}^2}{R q \bar{\rho}} \frac{dq}{d\bar{\rho}}
```

Comment:
The code use `d/dlog(rho)`, which makes it difficult to compare to the documentation. Given the lack of tests, this doesn't give me much confidence in the code

Reply:
d/dlog(rho) is the same as 1/rho d/d(rho) which I thought was clearer in the dox.  But I can put in explicit logs.